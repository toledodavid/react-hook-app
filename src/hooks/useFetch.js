import { useEffect, useState } from 'react';


export const useFetch = ( url ) => {
  const [state, setState] = useState({data: null, loading: true, error: null});
  
  useEffect(() => {
    fetch(url).then(response => response.json()).then(data =>{
      setState({
        loading: false,
        error: null,
        data
      });
    });
  }, [url]);

  return state;
}