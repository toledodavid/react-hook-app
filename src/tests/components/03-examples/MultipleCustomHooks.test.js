import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import MultipleCustomHooks from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');



describe('Tests in <MultipleCustomHooks /> component', () => {

  beforeEach(() => {
  
    useCounter.mockReturnValue({
      counter: 1,
      increment: () => {}
    });
  
  });

  test('It should render correctly', () => {

    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null
    });

    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });

  test('It should show information', () => {
    
    useFetch.mockReturnValue({
      data: [{
        author: 'David Toledo',
        quote: 'Hello world'
      }],
      loading: false,
      error: null
    });

    const wrapper = shallow(<MultipleCustomHooks/>);
    
    expect(wrapper.find('.alert').exists()).toBe(false);
    expect(wrapper.find('.mb-0').text().trim()).toBe('Hello world');
    expect(wrapper.find('footer').text().trim()).toBe('David Toledo');
  });
});