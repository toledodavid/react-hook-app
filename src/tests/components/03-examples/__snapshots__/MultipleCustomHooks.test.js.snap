// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <MultipleCustomHooks /> component It should render correctly 1`] = `
<div>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</div>
`;
